### PR TITLE
test: LEAP-1141: Add test for concurrent saving annotation and draft

### PR DESCRIPTION
**Pull request by Gondragos**
### PR fulfills these requirements
- [x] Commit message(s) and PR title follows the format `[fix|feat|ci|chore|doc]: TICKET-ID: Short description of change made` ex. `fix: DEV-XXXX: Removed inconsistent code usage causing intermittent errors`
- [x] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

### Describe the reason for change
The case was not covered by tests.

### Does this PR introduce a breaking change?
- [ ] Yes, and covered entirely by feature flag(s)
- [ ] Yes, and covered partially by feature flag(s)
- [x] No
- [ ] Not sure (briefly explain the situation below)

### What level of testing was included in the change?
- [ ] e2e
- [x] integration
- [ ] unit

### Which logical domain(s) does this change affect?
`Tests`, `Submiting process`, `Annotation`, `Draft`

### CODE REVIEW

File: web/libs/frontend-test/src/helpers/LSF/ToolBar.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
+import { FF_DEV_1170 } from "@humansignal/frontend-test/feature-flags";
+import { LabelStudio } from "@humansignal/frontend-test/helpers/LSF/LabelStudio";
+import { FF_DEV_3873 } from "../../../../editor/src/utils/feature-flags";
+
 export const ToolBar = {
   get root() {
-    return cy.get(".lsf-topbar");
+    return LabelStudio.getFeatureFlag(FF_DEV_3873).then((isFFDEV3873) => {
```
Comment (hlomzik) [inline web/libs/frontend-test/src/helpers/LSF/ToolBar.ts:7]: can we safely switch to `.lsf-bottombar` and all tests will work? assuming that we will have `FF_DEV_3873` enabled all the time

File: web/libs/editor/tests/integration/e2e/drafts/submit.cy.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      .withAnnotation({ userGenerate: true, result: [] })
+      .withEventListener("submitDraft", () => {
+        return new Promise((resolve) => {
+          // initialize saving annotation exactly when request of saving draft should be sent to the server
+          ToolBar.submitBtn.click();
+
+          // this emulates server response delay
+          setTimeout(() => {
+            callOrder.push(RESPONSE_TYPE.DRAFT);
+            resolve(void 0);
```
Comment (hlomzik) [inline web/libs/editor/tests/integration/e2e/drafts/submit.cy.ts]: `void 0`? why?
can we just skip this param?

Comment (Gondragos) [reply web/libs/editor/tests/integration/e2e/drafts/submit.cy.ts]: Idk. My IDE shows warning if there is nothing. 😅

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9673271180)
> Successfully merged:  create mode 100644 web/libs/editor/src/assets/icons/models.svg

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9686348321)
> Successfully merged:  15 files changed, 46 insertions(+), 26 deletions(-)

File: web/libs/editor/tests/integration/e2e/drafts/submit.cy.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+          // this emulates server response delay
+          setTimeout(() => {
+            callApi(RESPONSE_TYPE.DRAFT);
+            resolve();
+          }, RESPONSE_DELAY);
+        });
+      })
+      .withEventListener("submitAnnotation", () => {
+        callApi(RESPONSE_TYPE.SUBMIT);
+      })
```
Comment (hlomzik) [inline web/libs/editor/tests/integration/e2e/drafts/submit.cy.ts]: I'd suggest to switch methods for better readability: we assign handler for submit, then we assign handler for draft and trigger submit at the beginning of this handler

```suggestion
      .withEventListener("submitAnnotation", () => {
        callApi(RESPONSE_TYPE.SUBMIT);
      })
      .withEventListener("submitDraft", () => {
        return new Promise<void>((resolve) => {
          // initialize saving annotation exactly when request of saving draft should be sent to the server
          ToolBar.submitBtn.click();

          // this emulates server response delay
          setTimeout(() => {
            callApi(RESPONSE_TYPE.DRAFT);
            resolve();
          }, RESPONSE_DELAY);
        });
      })
```

**Review by hlomzik — APPROVED** [review]

File: web/libs/frontend-test/src/helpers/LSF/ToolBar.ts (hunk 1 above)
Comment (hlomzik) [reply web/libs/frontend-test/src/helpers/LSF/ToolBar.ts:7]: I'll address this [in my tests](https://github.com/HumanSignal/label-studio/pull/6019)

**Review by juliosgarbi — APPROVED** [review]

**Review by AndrejOros — APPROVED** [review]

**Comment by Gondragos** [comment]
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/9744672894)
> Successfully merged:  create mode 100644 docs/themes/v2/source/images/prompts/tuning-diagram.png

**Pull request merged**